src/Scrutor/AttributeSelector.cs

```diff
@@ -40,30 +40,35 @@ void ISelector.Populate(IServiceCollection services)
 
                 foreach (var attribute in attributes)
                 {
-                    var serviceType = GetServiceType(type, attribute);
+                    var serviceTypes = GetServiceType(type, attribute);
 
-                    var descriptor = new ServiceDescriptor(serviceType, type, attribute.Lifetime);
+                    foreach (var serviceType in serviceTypes)
+                    {
+                        var descriptor = new ServiceDescriptor(serviceType, type, attribute.Lifetime);
 
-                    services.Add(descriptor);
+                        services.Add(descriptor);
+                    }
                 }
             }
         }
 
-        private static Type GetServiceType(Type type, ServiceDescriptorAttribute attribute)
+        private static IEnumerable<Type> GetServiceType(Type type, ServiceDescriptorAttribute attribute)
         {
             var serviceType = attribute.ServiceType;
 
             if (serviceType == null)
             {
-                return type;
+                return type.GetInterfaces()
+                    .Concat(new [] {type, type.GetTypeInfo().BaseType})
```

Comment:
`BaseType` might be `null`. I'd rather check that first. That's an easy change after moving to an iterator method.